magic-links/ongoing/ongoing.js

```diff
@@ -0,0 +1,508 @@
+"use strict";
+
+let current_time_zone = Intl.DateTimeFormat().resolvedOptions().timeZone || 'America/Chicago'
+const now = new Date().getTime()/1000;
+let calendar_subscribe_object = {
+  start_timestamp: 0,
+  end_timestamp: 0,
+  slot_length: 15,
+  duration_options: {},
+  coverage_by_month: {}
+}
+let escapeObject = (obj) => {
+  return Object.fromEntries(Object.entries(obj).map(([key, value]) => {
+    return [ key, window.lodash.escape(value)]
+  }))
+}
+
+jQuery(document).ready(async function ($) {
+  let jsObject = window.campaign_objects
+
+  let data = await get_campaign_data()
+  $('.cp-wrapper').removeClass("loading-content")
+  $('.cp-loading-page').hide()
+
+  calendar_subscribe_object = { ...calendar_subscribe_object, ...data }
+  calendar_subscribe_object.translations = escapeObject( jsObject.translations )
+  let days = window.campaign_scripts.calculate_day_times()
+
+  let week_day_names = window.campaign_scripts.days_for_locale(navigator.language, 'narrow')
+  let headers = `
+    <div class="day-cell week-day">${week_day_names[0]}</div>
+    <div class="day-cell week-day">${week_day_names[1]}</div>
+    <div class="day-cell week-day">${week_day_names[2]}</div>
+    <div class="day-cell week-day">${week_day_names[3]}</div>
+    <div class="day-cell week-day">${week_day_names[4]}</div>
+    <div class="day-cell week-day">${week_day_names[5]}</div>
+    <div class="day-cell week-day">${week_day_names[6]}</div>
+  `;
+  let daily_time_select = $('#cp-daily-time-select')
+  let selected_times = [];
+  let current_time_selected = $("#cp-individual-time-select").val();
+  let modal_calendar = $('#day-select-calendar')
+
+  // coverage_by_month();
+
+  draw_calendar()
+
+  setup_signup();
+
+  update_timezone()
+
+  //draw progress circles
+  window.customElements.define('progress-ring', ProgressRing);
+
+  $('#confirm-timezone').on('click', function (){
+    current_time_zone = $("#timezone-select").val()
+    update_timezone()
+    days = window.campaign_scripts.calculate_day_times(current_time_zone)
+    // set_campaign_date_range_title()
+    populate_daily_select()
+    draw_calendar()
+    draw_modal_calendar()
+  })
+
+
+  function setup_signup(){
+    if ( calendar_subscribe_object.status === "inactive"){
+      $('#cp-view-closed').show()
+      $("#cp-wrapper").css("min-height", '500px')
+    } else {
+      $('#cp-main-page').show()
+    }
+
+    //navigation function
+    $('.cp-nav').on( 'click', function (){
+      $('.cp-view').hide()
+      let view_to_open = $(this).data('open')
+      $(`#${view_to_open}`).show()
+
+      //force the screen to scroll to the top of the wrapper
+      if ( $(this).data('force-scroll')){
+        let elmnt = document.getElementById("cp-wrapper");
+        elmnt.scrollIntoView();
+      }
+
+      //configure the view to go back to
+      let back_to = $(this).data('back-to');
+      if ( back_to )
+      $(`#${view_to_open} .cp-close-button`).data('open', back_to)
```

Comment:
Totally minor, but I guess on the point of code style, are we happy with inline control statements in the JS?
I know the phpcs complains about it, and looking at these 2 lines I can see why it might not be a good idea...

Reply:
Tada :)
Yeah. Not our usual